### DISCUSSION TITLE
Allow for setting cursor location in TextInput

### DISCUSSION THREAD

**Discussion by Derrreks** [mantinedev/mantine#1507] [Feature requests]
We have some custom input components that wrap Mantine's `TextInput` - for instance, a component that formats/better renders a phone number as the user types it in (adding parens and dashes in the right places). However, when we rewrite the `input`'s value it resets the cursor to the end, leaving a weird experience if a user tries to edit the middle.

On a base HTML `input` component, one might solve this using the `setSelectionRange` function ([docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange)). From my reading of the Mantine docs, it does not seem possible to use this or any similar alternative with `TextInput`.

Am I mistaken and this _is_ possible? Alternatively, might this be an interesting addition to Mantine?

**Comment by rtivital** [comment]
TextInput renders regular input element, you can perform any actions as you would do with a regular input element
